odxtools/database.py

```diff
@@ -61,21 +63,25 @@ def __init__(self,
         for root in documents:
             # ODX spec version
             model_version = version(root.attrib.get("MODEL-VERSION", "2.0"))
+            if self.model_version is not None and self.model_version != model_version:
+                odxraise(f"Different ODX versions used in the same file (ODX {model_version} "
+                         f"and ODX {self.model_version}")
+            self.model_version = model_version
             dlc = root.find("DIAG-LAYER-CONTAINER")
             if dlc is not None:
                 dlcs.append(DiagLayerContainer.from_et(dlc, []))
-            # In ODX 2.0 there was only COMPARAM-SPEC
-            # In ODX 2.2 content of COMPARAM-SPEC was renamed to COMPARAM-SUBSET
-            # and COMPARAM-SPEC becomes a container for PROT-STACKS
-            # and a PROT-STACK references a list of COMPARAM-SUBSET
-            if model_version >= version("2.2"):
-                subset = root.find("COMPARAM-SUBSET")
-                if subset is not None:
-                    comparam_subsets.append(ComparamSubset.from_et(subset, []))
-            else:
-                spec = root.find("COMPARAM-SPEC")
-                if spec is not None:
-                    comparam_specs.append(ComparamSpec.from_et(spec, []))
+
+            # In ODX 2.0 there was only COMPARAM-SPEC. In ODX 2.2 the
+            # content of COMPARAM-SPEC was moved to COMPARAM-SUBSET
+            # and COMPARAM-SPEC became a container for PROT-STACKS and
+            # a PROT-STACK references a list of COMPARAM-SUBSET
+            cp_subset = root.find("COMPARAM-SUBSET")
+            if cp_subset is not None:
+                comparam_subsets.append(ComparamSubset.from_et(cp_subset, []))
+
+            cp_spec = root.find("COMPARAM-SPEC")
+            if cp_spec is not None:
+                comparam_specs.append(ComparamSpec.from_et(cp_spec, []))
```

Comment:
This may look strange at first, but between 2.0 and 2.2 only the XML tag changed, the content of the COMPARAM-SPEC in 2.0 is basically a COMPARAM-SUBSET in 2.2
This way the code needed for 2.0 support remains minimal
```suggestion
                comparam_specs.append(ComparamSubset.from_et(cp_spec, []))
```

Reply:
ah, I see. it's better to append these objects to `comparam_subsets` then, though? (This would also make conversion of datasets from odx 2.0 to 2.2 easier.) I've implemented it this way for now: adf3dfb946eb35519f866282df0f90b369bd0c97